buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -59,24 +58,23 @@
     private final String clusterName;
     private final NamedDomainObjectContainer<ElasticsearchNode> nodes;
     private final File workingDirBase;
-    private final Function<Integer, ElasticsearchDistribution> distributionFactory;
     private final LinkedHashMap<String, Predicate<TestClusterConfiguration>> waitConditions = new LinkedHashMap<>();
     private final Project project;
     private final ReaperService reaper;
+    private int nodeIndex  = 0;
 
-    public ElasticsearchCluster(String path, String clusterName, Project project, ReaperService reaper,
-                                Function<Integer, ElasticsearchDistribution> distributionFactory, File workingDirBase) {
+    public ElasticsearchCluster(String path, String clusterName, Project project,
+                                ReaperService reaper, File workingDirBase) {
```

Comment:
I think we should try and keep the distribution factory pattern here rather than have the `ElasticsearchNode` directly call into the `DistributionContainer`.

Reply:
Could you expand on the advantages you see here ? To me it seemed like indirection that makes things harder to understand. We have proffered explicit coupling elsewhere.

Reply:
We should avoid this kind of coupling when at all possible. That's the reason for this factory pattern here initially. I think it's actually easier to understand with that factory logic living in the plugin. It removes some noise from the `ElasticsearchNode` class which is already getting quite complex and should realistically be broken up at some point.